### DISCUSSION TITLE
How do you delay a save_screeshot, or making it exact on a certain page?

### DISCUSSION THREAD

**Discussion by Uno-count** [seleniumbase/SeleniumBase#2618] [Q&A] [answered]
In my situation, the save_screenshot function occasionally captures a page that has not fully loaded, resulting in an incomplete elements in the page.
![image](https://github.com/seleniumbase/SeleniumBase/assets/73009596/16bf5a46-51b4-47d4-8a2d-48b328af1574)

**Comment by mdmintz** [comment, marked as answer]
Assuming you haven't modified the default `pageLoadStrategy` for SeleniumBase, methods will wait for the `readyState` of the web page to be `complete` before performing actions, such as saving a screenshot with `self.save_screenshot("NAME.png")`.

You can also wait for a specific element to appear first by using `self.wait_for_element(selector)`.
